spec/models/contribution_spec.rb

```diff
@@ -107,4 +107,34 @@
       expect(described_class.without_pr_state(:merged)).to contain_exactly(open_pr)
     end
   end
+
+  describe 'validations' do
+    context 'validate rejected_reason_presence' do
+      context 'when the contribution is refused' do
+        let(:contribution) { create(:contribution) }
+
+        before { contribution.update(state: :refused) }
+        it 'has a rejected reason' do
+          expect { contribution.update(rejected_reason: :allocated_in_the_project) }.to change { contribution.rejected_reason }.from(nil).to('allocated_in_the_project')
+        end
+      end
+
+      context 'when the contribution is received' do
+        let(:contribution) { create(:contribution) }
+
+        it 'raises an error' do
+          expect { contribution.update!(rejected_reason: :allocated_in_the_project) }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
You should also verify the error message here the same way you did with approved context

Reply:
Sure! Already made the changes